codespeed/views.py

```diff
@@ -731,3 +739,239 @@ def add_json_results(request):
+        return None, HttpResponseNotFound(
+                        'Executable "' + data['exe'] + '" does not exist!')
+    except Branch.DoesNotExist:
+        return None, HttpResponseNotFound(
+                        'Branch "' + data['branch'] + '" does not exist!')
+    except Benchmark.DoesNotExist:
+        return None, HttpResponseNotFound(
+                        'Benchmark "' + data['ben'] + '" does not exist!')
+
+    number_of_revs = int(data.get('revs', 10))
+
+    baseline_commit_name = (data['base_commit'] if 'base_commit' in data
+                            else None)
+    relative_results = (
+        ('relative' in data and data['relative'] in ['1', 'yes']) or
+        baseline_commit_name is not None)
+
+    result_query = Result.objects.filter(
+        benchmark=benchmark
+    ).filter(
+        environment=environment
+    ).filter(
+        executable=executable
+    ).filter(
+        revision__project=project
+    ).filter(
+        revision__branch=branch
+    ).select_related(
+        "revision"
+    ).order_by('-date')[:number_of_revs]
+
+    if len(result_query) == 0:
+        return None, HttpResponseNotFound("No results were found!")
+
+    result_list = [item for item in result_query]
+    result_list.reverse()
+
+    if relative_results:
+        ref_value = result_list[0].value
+
+    if baseline_commit_name is not None:
+        baseline_env = environment
+        baseline_proj = project
+        baseline_exe = executable
+        baseline_branch = branch
+
+        try:
+            if 'base_env' in data:
+                baseline_env = Environment.objects.get(name=data['base_env'])
+            if 'base_proj' in data:
+                baseline_proj = Project.objects.get(name=data['base_proj'])
+            if 'base_exe' in data:
+                baseline_exe = Executable.objects.get(name=data['base_exe'],
+                                                      project=baseline_proj)
+            if 'base_branch' in data:
+                baseline_branch = Branch.objects.get(name=data['base_branch'],
+                                                     project=baseline_proj)
+        except Environment.DoesNotExist:
+            return None, HttpResponseNotFound(
+                            'Baseline environment "' + data['base_env'] +
+                            '" does not exist!')
+        except Project.DoesNotExist:
+            return None, HttpResponseNotFound(
+                            'Baseline project "' + data['base_proj'] +
+                            '" does not exist!')
+        except Executable.DoesNotExist:
+            return None, HttpResponseNotFound(
+                            'Baseline executable "' + data['base_exe'] +
+                            '"does not exist!')
+        except Branch.DoesNotExist:
+            return None, HttpResponseNotFound(
+                            'Baseline branch "' + data['base_branch'] +
+                            '" does not exist!')
```

Comment:
Instead of returning `None, err` all the time, we could return just the data or raise a django ObjectDoesNotExist exception (with the same message being set now). The use of the function (the view) could then catch the error and return the HttpResponseNotFound.